### PR TITLE
Feature/s3 c 2798 get bucket notification

### PR DESCRIPTION
**Pull request by dora-korpar**
Adds `getBucketNotification` API.
Requires https://github.com/scality/cloudserver/pull/2946 and https://github.com/scality/Arsenal/pull/1235 to be merged.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -77,5 +77,14 @@
         "viaProxy": false,
         "trustedProxyCIDRs": [],
         "extractClientIPFromHeader": ""
-    }
+    },
+    "bucketNotificationTargets": [
```
Comment (naren-rajendran) [inline config.json]: The property is renamed to bucketNotificationDestinations, ref https://github.com/scality/Federation/blob/4bf06519734ea45479ecb60dbd0a1cb86e4223a9/roles/run-s3/templates/config.json.j2#L139

**Review by naren-rajendran — COMMENTED** [review]
Just the property name change, otherwise looks good to me.

**Review by naren-rajendran — APPROVED** [review]

**Review by anurag4DSB — APPROVED** [review]

**Comment by dora-korpar** [comment]
@bert-e reset

File: lib/api/bucketGetNotification.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+const { metadataValidateBucket } = require('../metadata/metadataUtils');
+const { pushMetric } = require('../utapi/utilities');
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+const { NotificationConfiguration } = require('arsenal').models;
+
+/**
+ * Format of xml response:
+ *
+ * <ONotificationConfiguration>
```
Comment (miniscruff) [inline lib/api/bucketGetNotification.js]: Extra O at the start
```suggestion
 * <NotificationConfiguration>
```

File: tests/functional/aws-node-sdk/test/bucket/getBucketNotification.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+describe('aws-sdk test get bucket notification', () => {
+    let s3;
+    let otherAccountS3;
+
+    before(done => {
+        const config = getConfig('default', { signatureVersion: 'v4' });
+        s3 = new S3(config);
+        otherAccountS3 = new BucketUtility('lisa', {}).s3;
+        return done();
+    });
```
Comment (miniscruff) [inline tests/functional/aws-node-sdk/test/bucket/getBucketNotification.js]: no need for done.
```suggestion
    before(() => {
        const config = getConfig('default', { signatureVersion: 'v4' });
        s3 = new S3(config);
        otherAccountS3 = new BucketUtility('lisa', {}).s3;
    });
```

File: tests/unit/api/bucketGetNotification.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        `<QueueArn>${queueArn2}</QueueArn>` +
+        '</QueueConfiguration>' +
+        '</NotificationConfiguration>';
+}
+
+
+describe('getBucketNotification API', () => {
+    before(() => cleanup());
+    beforeEach(done => bucketPut(authInfo, testBucketPutRequest, log, done));
+    afterEach(() => cleanup());
```
Comment (miniscruff) [inline tests/unit/api/bucketGetNotification.js]: No need for extra function.
```suggestion
    before(cleanup);
    beforeEach(done => bucketPut(authInfo, testBucketPutRequest, log, done));
    afterEach(cleanup);
```

File: tests/unit/api/bucketGetNotification.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            done();
+        });
+    });
+
+    describe('after bucket notification has been put', () => {
+        beforeEach(done => {
+            const putRequest =
+                getNotificationRequest(bucketName, getNotificationXml());
+            bucketPutNotification(authInfo, putRequest, log, err => {
+                assert.equal(err, null);
```
Comment (miniscruff) [inline tests/unit/api/bucketGetNotification.js]: every other assert uses ifError, is this different for a reason?

File: tests/functional/aws-node-sdk/test/bucket/getBucketNotification.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+// Check for the expected error response code and status code.
+function assertError(err, expectedErr, cb) {
+    if (expectedErr === null) {
+        assert.strictEqual(err, null, `expected no error but got '${err}'`);
+    } else {
+        assert.strictEqual(err.code, expectedErr, 'incorrect error response ' +
+            `code: should be '${expectedErr}' but got '${err.code}'`);
+        assert.strictEqual(err.statusCode, errors[expectedErr].code,
+            'incorrect error status code: should be 400 but got ' +
+            `'${err.statusCode}'`);
```
Comment (miniscruff) [inline tests/functional/aws-node-sdk/test/bucket/getBucketNotification.js]: Not sure I would add extra messages here as the asserted error will already output something very similar to this.

File: tests/unit/api/bucketGetNotification.js (hunk 5 above)
Comment (dora-korpar) [reply tests/unit/api/bucketGetNotification.js]: copy pasta mistake

**Comment by dora-korpar** [comment]
@bert-e reset

**Comment by dora-korpar** [comment]
@bert-e approve

**Pull request merged**